File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> result;
    void print_subseq(vector<int> v,int i,vector<int>& nums){
        if(i>=nums.size()){
            result.push_back(v);
            return;
        }

        v.push_back(nums[i]);
        print_subseq(v,i+1,nums);

        v.pop_back();
        print_subseq(v,i+1,nums);

    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int> temp;
        print_subseq(temp,0,nums);
        return result;
    }
};

File: 0078-subsets/0078-subsets.py
class Solution(object):
    def solve(self,nums,i,subset,res):
        if(i==len(nums)):
            res.append(list(subset))
            return
        #print(subset)
        self.solve(nums,i+1,subset,res)
        subset.append(nums[i])
        self.solve(nums,i+1,subset,res)
        subset.pop()

    def subsets(self, nums):
        i = 0
        subset = []
        res = []
        self.solve(nums,i,subset,res)
        return res

        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        